src/main/java/picard/analysis/CollectQualityYieldMetricsSNVQ.java

```diff
@@ -0,0 +1,548 @@
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SAMTag;
+import htsjdk.samtools.metrics.MetricsFile;
+import htsjdk.samtools.reference.ReferenceSequence;
+import htsjdk.samtools.util.Histogram;
+import htsjdk.samtools.util.IOUtil;
+import org.broadinstitute.barclay.argparser.Argument;
+import org.broadinstitute.barclay.argparser.CommandLineProgramProperties;
+import org.broadinstitute.barclay.help.DocumentedFeature;
+import picard.PicardException;
+import picard.cmdline.StandardOptionDefinitions;
+import picard.cmdline.programgroups.DiagnosticsAndQCProgramGroup;
+import picard.util.SeriesStats;
+import picard.util.help.HelpConstants;
+
+import java.io.File;
+import java.util.Vector;
+
+/**
+ * Command line program to calculate SNV quality yield metrics for read files
+ *
+ * @author Dror Kessler
+ */
+
+@CommandLineProgramProperties(
+        summary = CollectQualityYieldMetricsSNVQ.USAGE_SUMMARY + CollectQualityYieldMetricsSNVQ.USAGE_DETAILS,
+        oneLineSummary = CollectQualityYieldMetricsSNVQ.USAGE_SUMMARY,
+        programGroup = DiagnosticsAndQCProgramGroup.class
+)
+@DocumentedFeature
+public class CollectQualityYieldMetricsSNVQ extends SinglePassSamProgram {
+    private QualityYieldMetricsCollector collector = null;
+    public Histogram<Integer> qualityHistogram = new Histogram<>("KEY", "BQ_COUNT");
+    private Vector<SeriesStats> readPositionQualityStats = new Vector<>();
+    public Histogram<Integer> snvqHistogram = new Histogram<>("KEY", "SNVQ_COUNT");
+    private Vector<SeriesStats> readPositionSnvqStats = new Vector<>();
+    final byte[] baseOrder = "ACGT".getBytes();
+
+    static final String USAGE_SUMMARY = "Collect SNVQ metrics about reads that pass quality thresholds and Illumina-specific filters.  ";
+    static final String USAGE_DETAILS = "This tool evaluates the overall SNVQ quality of reads within a bam file containing one read group. " +
+            "The output indicates the total numbers of bases within a read group that pass a minimum base quality score threshold and " +
+            "(in the case of Illumina data) pass Illumina quality filters as described in the " +
+            "<a href='https://www.broadinstitute.org/gatk/guide/article?id=6329'>GATK Dictionary entry</a>. " +
+            "<br />" +
+
+            "<h4>Usage Example:</h4>" +
+            "<pre>" +
+            "java -jar picard.jar CollectSNVQualityYieldMetrics \\<br /> " +
+            "      I=input.bam \\<br /> " +
+            "      O=quality_yield_metrics.txt \\<br />" +
+            "</pre>" +
+            "Please see " +
+            "<a href='https://broadinstitute.github.io/picard/picard-metric-definitions.html#CollectSNVQualityYieldMetrics.QualityYieldMetrics'>" +
+            "the QualityYieldMetrics documentation</a> for details and explanations of the output metrics." +
+            "<hr />";
+
+    @Argument(shortName = StandardOptionDefinitions.USE_ACTUAL_BASE_QUALITIES_SHORT_NAME,
```

Comment:
Do you mean that this allows to calculate basequality when the qualities are in the flow space? It is a bit confusing I think. Maybe better interface would be 
1. ApplySNVQ by default replaces the base qualities
2. There is an option to take qualities from arbitrary tag (BQ by default? not 100% sure)

Reply:
@dror27 - do we need to discuss this one?

Reply:
This has been transitioned to USE_BQ_FOR_BASE_QUALITIES. By default it is false, meaning the QUAL string is used for base qualities. If true, then BQ is used.